src/inpod/statemanager.rs

```diff
@@ -233,10 +237,18 @@ impl WorkloadProxyManagerState {
         let workload_netns_inode = netns.workload_inode();
         let (drain_tx, drain_rx) = drain::channel();
 
+        // this matches istio's generatePodUID
```

Comment:
> UID represents a globally unique **opaque** identifier for this workload.
> For k8s resources, it is **recommended** to use the more readable format:

Assuming a format is not valid here

Reply:
yes, I'm aware - see my questions in the PR body.  TL,DR I think we have these options:

- Computing the UID this way (with implied dep with ambient index)
- Support fetching with the pod uid in the ambient index, so that ztunnel can fetch workloads by it
- Change the node agent to send ambient-index style uuids instead of pod UIDs.

Reply:
Hi folks, can we drive some resolution here?  Which one is best - pod UID or UUID?

Reply:
hi @howardjohn any thoughts on above options?

Reply:
Option 2 basically means that each workload has multiple UID's. So while we don't depend on the format of one UID, we now have a _new_ UID and we _do_ depend on that UID's format. That doesn't feel great.

Is it crazy to not use UID at all? We could match on quite a few options.. some combination of `{node+cluster, IP, SA, name, namespace}`? We may not need to assert _all_ of them, though we could.

Reply:
Just to clarify i mean to pass this info through PodInfo. The node+cluster is not needed in PodInfo, ztunnel knows its own node/cluster.

Reply:
> Is it crazy to not use UID at all? We could match on quite a few options.. some combination of {node+cluster, IP, SA, name, namespace}? We may not need to assert all of them, though we could.

Does that really save much in terms of the workload query API tho? That's still basically just a UID lookup but with a local key schema/alias.

Reply:
Putting Pod UID as an alias seems like "We had a UID, but the spec said its opaque so we built a _new_ UID field and treat that as non-opaque". Now we have 2 UIDs for one resource, and we still have non-opaque UID. It feels like the worst of both worlds.

Reply:
Pod UID is the best thing to use on as it covers all edge cases (like pod being re-used with same name but different service account).

I'm not sure what you mean by "match"? - My understanding is that whatever property we use needs to be a valid resource name so it can work with on-demand WDS?

I agree that having 2 uids is not great. why did we decide that the resource name is a UID though? i think it's very reasonable that the resource name would have a semantic meaning?

Reply:
> Putting Pod UID as an alias seems like "We had a UID, but the spec said its opaque so we built a new UID field and treat that as non-opaque". Now we have 2 UIDs for one resource, and we still have non-opaque UID. It feels like the worst of both worlds.

I think everyone is in agreement on this point, but since I'm the new guy here - why is UID opacity important here, exactly? What led to that requirement?

Reply:
> I'm not sure what you mean by "match"? - My understanding is that whatever property we use needs to be a valid resource name so it can work with on-demand WDS?

You can lookup by IP then just verify it matches all the properties (`{node+cluster, IP, SA, name, namespace}`)?

The lookup logic is already pretty complex, I am worried about making it so we have so many aliases and lookup patterns for each resource.

Reply:
Should we just drop `fetch_workload_by_uid` from `state.rs` then?

If we don't want clients using the non-opaque internal UID format, probably shouldn't expose it at all.

(and if we don't clients people using it, why does it matter if it's opaque or not?)

Reply:
TBH I did not know that UID lookup was added. If we do go with the approach I mentioned I think we could drop that